### PR TITLE
chore: [ANDROSDK-1860] Simplify and automatize release process

### PR DESCRIPTION
**Pull request by vgarciabnz**
Changes the release process:

- Modify the Jenkinsfile to remove the publication in `master` (snapshot publication in `develop` is still enabled).
- Apply to plugin Gradle Nexus Publish to avoid the manual approval in Sonatype dashboard. This was a required step to do before any release publication.
- Create a github action to trigger snapshot or release publication (depending on the branch). It is the only way to publish release publications.

The release process is explained [in the wiki](https://github.com/dhis2/dhis2-android-sdk/wiki/Release-process), take a look and comment anything that is not clear.

This PR is open against `master` because the github action needs to be present in the default branch in order to be triggered in the "Actions" dashboard.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          fetch-depth: 0
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: temurin
+          cache: 'gradle'
+      - name: Change wrapper permission
+        run: chmod +x ./gradlew
+
```
[the remark this answers is not in the document]
Comment (andresmr) [reply .github/workflows/release.yml:20]: mm not sure, I don't have a clear opinion. I think is clear using a script maybe avoid duplicate code:

```
if [ "$branch" = "main" ]; then
  ./gradlew :designsystem:publishToSonatype closeAndReleaseSonatypeStagingRepository -PremoveSnapshotSuffix
else
  ./gradlew :designsystem:publishToSonatype
fi
```
Also in the action you can set specific private keys...

File: buildSrc/src/main/kotlin/maven-publish-conventions.gradle.kts [hunk 2]
```diff
@@ -114,22 +79,12 @@ afterEvaluate {
...
-        signing {
-            setRequired({ isReleaseBuild() && gradle.taskGraph.hasTask("publishing") })
-            sign(publishing.publications)
-        }
+    signing {
+        setRequired({ isReleaseBuild() && gradle.taskGraph.hasTask("publishing") })
+        useInMemoryPgpKeys(signingPrivateKey, signingPassword)
+        sign(publishing.publications)
     }
-}
```
Comment (andresmr) [inline buildSrc/src/main/kotlin/maven-publish-conventions.gradle.kts:135]: don't you need?
```
// Fix Gradle warning about signing tasks using publishing task outputs without explicit dependencies
// https://github.com/gradle/gradle/issues/26091
tasks.withType<AbstractPublishToMaven>().configureEach {
    val signingTasks = tasks.withType<Sign>()
    mustRunAfter(signingTasks)
}
```

**Review by andresmr — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vgarciabnz) [reply .github/workflows/release.yml:20]: yeah, I will change it

File: buildSrc/src/main/kotlin/maven-publish-conventions.gradle.kts (hunk 2 above)
Comment (vgarciabnz) [reply buildSrc/src/main/kotlin/maven-publish-conventions.gradle.kts:135]: This is not needed because it has not a multiplatform structure.

**Pull request merged**